src/main/java/tds/exam/results/repositories/impl/ExamReportAuditRepositoryImpl.java

```diff
@@ -38,24 +44,61 @@
     }
 
     @Override
-    public void insertExamReport(final UUID examId, final String examReportXml) {
+    public void insertExamReport(final UUID examId, final String examReportXml, final ExamReportStatus status) {
         final SqlParameterSource parameters = new MapSqlParameterSource("examId", examId.toString())
             .addValue("examReportXml", examReportXml)
-            .addValue("createdAt", mapInstantToTimestamp(Instant.now()));
+            .addValue("createdAt", mapInstantToTimestamp(Instant.now()))
+            .addValue("status", status.getValue());
 
         final String SQL =
             "INSERT INTO \n" +
                 "   exam_report \n" +
                 "( \n" +
                 "   exam_id, \n" +
                 "   report, \n" +
+                "   status, \n" +
                 "   created_at \n" +
                 ") VALUES ( \n" +
                 "   :examId, \n" +
                 "   :examReportXml, \n" +
+                "   :status, \n" +
                 "   :createdAt \n" +
                 ")";
 
         jdbcTemplate.update(SQL, parameters);
     }
+
+    @Override
+    public Optional<ExamReport> findLatestExamReport(final UUID examId) {
+        final SqlParameterSource parameters = new MapSqlParameterSource("examId", examId.toString());
+
+        final String SQL = "SELECT \n" +
+            "  report, \n" +
+            "  exam_id, \n" +
+            "  status \n" +
+            "FROM \n " +
+            "  exam_report \n" +
+            "WHERE \n" +
+            "  exam_id = :examId \n" +
+            "ORDER BY id DESC LIMIT 1";
+
+        Optional<ExamReport> maybeExamReport;
+
+        try {
+            maybeExamReport = Optional.of(jdbcTemplate.queryForObject(SQL, parameters, new RowMapper<ExamReport>() {
```

Comment:
Nothing wrong with this code at all, but I think you can you use a lambda here to instantiate your RowMapper if you wanted to, which I personally find a little cleaner that creating/overriding the functional interface's single method.